File: app/src/main/java/com/blues/framework/base/BaseViewPagerAdapter.java
package com.blues.framework.base;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * User : Blues
 * Date : 2019/6/27
 * Time : 11:26
 */

public class BaseViewPagerAdapter extends FragmentPagerAdapter {

    private String[] titles;
    private List<Fragment> fragments = new ArrayList<>();

    public BaseViewPagerAdapter(FragmentManager fm, String[] titles) {
        super(fm);
        this.titles = titles;
    }

    @Override
    public Fragment getItem(int i) {
        return fragments.get(i);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    public void addFragment(Fragment fragment) {
        fragments.add(fragment);
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }
}

File: app/src/main/java/com/blues/framework/base/BaseViewPagerAdapter.kt
package com.blues.framework.base

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import java.util.ArrayList

/**
 * User : Blues
 * Date : 2019/6/27
 * Time : 11:26
 */
class BaseViewPagerAdapter(fm: FragmentManager?, private val titles: List<String>) :
    FragmentPagerAdapter(fm!!) {

    private val fragments: MutableList<Fragment> = ArrayList()

    override fun getItem(i: Int): Fragment {
        return fragments[i]
    }

    override fun getCount(): Int {
        return fragments.size
    }

    fun addFragment(fragment: Fragment) {
        fragments.add(fragment)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titles[position]
    }
}